test/functional/feature_block.py

```diff
@@ -137,20 +137,24 @@ def run_test(self):
         for TxTemplate in invalid_txs.iter_all_templates():
             template = TxTemplate(spend_tx=attempt_spend_tx)
 
-            # Something about the serialization code for missing inputs creates
-            # a different hash in the test client than on bitcoind, resulting
-            # in a mismatching merkle root during block validation.
-            # Skip until we figure out what's going on.
-            if TxTemplate == invalid_txs.InputMissing:
-                continue
             if template.valid_in_block:
                 continue
 
             self.log.info("Reject block with invalid tx: %s", TxTemplate.__name__)
             blockname = "for_invalid.%s" % TxTemplate.__name__
             badblock = self.next_block(blockname)
             badtx = template.get_tx()
-            self.sign_tx(badtx, attempt_spend_tx)
+            if TxTemplate != invalid_txs.InputMissing:
+                self.sign_tx(badtx, attempt_spend_tx)
+            else:
+               # Segwit is active in regtest at this point, so to deserialze a
+               # transaction without any inputs correctly, we set the outputs
+               # to an empty list. This is a hack, as the serialization of an
+               # empty list of outputs is deserialized as flags==0 and thus
+               # deserialization of the outputs is skipped.
+               # Though, it works for the purpose that both sides calculate the
+               # same tx hash in the merkle tree
```

Comment:
Not familiar with this test, but just reading the comment, it makes me wonder what outputs were previously in `badtx.vout`, and why they were generated to begin with if they can be discarded here. If it would make the hack slightly less hacky to avoid generating the outputs originally, or if there is a reason outputs just aren't important, it would be nice to state this in the comment.